examples/language-modeling/README.md

```diff
@@ -95,6 +95,59 @@ python ../gaudi_spawn.py \
 This takes about 4 minutes to train on 8 HPUs. It reaches
 a perplexity of 21.7968 once fine-tuned on the dataset.
 
+## GPT-J-6B and causal language modeling
+
+The following command triggers fine-tuning of GPT-J-6B model on WikiText-2 with Deepspeed ZeRO-2.
+Fine tuning on 8 HPU cards takes around 6 mins with batchsize of 32 (per device is 4).
+It reaches a perplexity of 14.011 once finetuned with the dataset.
+### Multi-card Training with Deepspeed
+
+```bash
+python ../gaudi_spawn.py \
+    --world_size 8 --use_deepspeed run_clm.py \
+    --model_name_or_path EleutherAI/gpt-j-6b \
+    --dataset_name wikitext \
+    --dataset_config_name wikitext-2-raw-v1 \
+    --per_device_train_batch_size 4 \
+    --per_device_eval_batch_size 4 \
+    --do_train \
+    --do_eval \
+    --output_dir /tmp/test-clm-xl-1 \
+    --gaudi_config_name Habana/gpt2 \
+    --use_habana \
+    --use_lazy_mode \
+    --gradient_checkpointing \
+    --use_hpu_graphs_for_inference \
+    --throughput_warmup_steps 2 \
+    --deepspeed path_for_deepspeed_config
+```
+## GPT-NEOX-20B and causal language modeling
+
+The following command triggers fine-tuning of GPT-NeoX-20B model on WikiText-2 with Deepspeed ZeRO-2.
+Fine tuning on 16 HPU cards takes around 9 mins with batch size of 32 (per device is 2).
+It reaches a perplexity of 10.469 once finetuned with the dataset.
+### Multi-Node Training with Deepspeed
+
+```bash
+python ../gaudi_spawn.py \
+    --hostfile path_for_the_hostfile \
```

Comment:
@regisss  : This doesn't work in single node /8 cards - we get mem allocation failures. 
It works with  a minimum 2 nodes/16 cards with batch size : 2, we would need more cards to increase the batch size.

Reply:
It should still work but with a total batch size of 16 instead of 32 no?
If you think it's important to keep the command line for 16 cards, feel free to revert adding a link to [this](https://github.com/huggingface/optimum-habana/tree/main/examples/multi-node-training) as it explains how to set up multi-node training.

Reply:
It did not work with per device batch size 2, i.e total batch size 16  in a single node